src/core/cgroup.c

```diff
@@ -3228,26 +3228,36 @@ static int cg_bpf_mask_supported(CGroupMask *ret) {
 
         /* BPF-based firewall */
         r = bpf_firewall_supported();
+        if (r < 0)
```

Comment:
@yuwata do you think there's a risk of people adding a new "is supported" function that returns `r < 0` for minor errors?

The way it's written now only fatal errors should return `r < 0` for "is supported" functions. But I'm concerned someone will accidentally return `r < 0` in the future for something like `EPERM`

Reply:
@anitazha Sorry for late response.
> do you think there's a risk of people adding a new "is supported" function that returns r < 0 for minor errors?

In that case, manage_new() fails, and we failed to boot system correctly. So, I hope such failure detected CIs and then we notice the direct regression.

But for safety, handling that `bpf_xxx_supported()` reported a negative errno as `not supported` may be also a good idea.
If you'd like, please submit a PR.